src/util/type_byte_size.h

```diff
@@ -7,21 +7,35 @@
 #include <util/namespace.h>
 #include <util/std_types.h>
 
-BigInt member_offset_bits(const type2tc &type, const irep_idt &member);
-BigInt member_offset(const type2tc &type, const irep_idt &member);
+BigInt member_offset_bits(
+  const type2tc &type,
+  const irep_idt &member,
+  const namespacet *ns = nullptr);
+BigInt member_offset(
+  const type2tc &type,
+  const irep_idt &member,
+  const namespacet *ns = nullptr);
 
 /* These can throw array_type2t::inf_sized_array_excp or
  * array_type2t::dyn_sized_array_excp */
-BigInt type_byte_size_bits(const type2tc &type);
-BigInt type_byte_size(const type2tc &type);
-BigInt type_byte_size_default(const type2tc &type, const BigInt &defaultval);
+BigInt type_byte_size_bits(const type2tc &type, const namespacet *ns = nullptr);
```

Comment:
I say we make the namespacet not default to nullptr and require it to always be passed to this functions (and related).

Reply:
It's quite a lot of places that use these functions. I won't mandate a change of signature in this PR. If this PR ever works.

Reply:
Since we are changing the behaviour of the function that will require the namespace, we should make it mandatory to pass the namespace. That way we can prevent clients from misusing the API, e.g., passing a symbol_typet but not passing the namespace.

Reply:
I don't understand. We are not changing the behaviour of any existing callers that use the default value for `ns`. They already know they can't pass symbol-types into these functions because that'll lead to abort(). And symbol-types should all be dealt with in thrash_type_symbols. No change here. If not all have been replaced or some part before thrash_type_symbols uses these functions, that's a semantic change of this PR we should catch, I agree. But not for those after. In some cases there not even is a namespace available, in simplify() for example. Changing each call site and by extension passing some(?) namespace to those that don't have one is too invasive for this PR.